src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -602,6 +602,17 @@ private void Dispose()
             DebuggedThread thread = await ThreadCache.GetThread(tid);
             await ThreadCache.StackFrames(thread);  // prepopulate the break thread in the thread cache
             ThreadContext cxt = await ThreadCache.GetThreadContext(thread);
+
+            if (cxt == null)
+            {
+                // Something went seriously wrong. For instance, this can happen when the primary thread
+                // of an app exits on linux while background threads continue to run with pthread_exit on the main thread
+                // See https://devdiv.visualstudio.com/DefaultCollection/DevDiv/VS%20Diag%20IntelliTrace/_workItems?_a=edit&id=197616&triage=true
+                // for a repro
+                Debug.Fail("Failed to find thread on break event.");
+                throw new MIException(Constants.E_FAIL);
```

Comment:
Sure, I can add something. I didn't see a lot of exceptions in mi that had more detailed error messages except for the miexception type which couldn't really be used here. I'll throw something with a message that describes that the thread couldn't be found.